Branch-SDK/src/main/java/io/branch/referral/network/BranchRemoteInterfaceUrlConnection.java

```diff
@@ -203,14 +207,23 @@ private BranchResponse doRestfulPost(String url, JSONObject payload, int retryNu
                 if (ex instanceof NetworkOnMainThreadException)
                     PrefHelper.Debug("Branch Error: Don't call our synchronous methods on the main thread!!!");
             }
-            return new BranchResponse(null, 500);
+            return new BranchResponse(null, 500, requestId);
         } finally {
             if (connection != null) {
                 connection.disconnect();
             }
         }
     }
 
+    @Nullable
+    private String getRequestIdFromHeader(@NonNull HttpsURLConnection connection) {
+        if (!connection.getHeaderFields().isEmpty() && connection.getHeaderFields()
```

Comment:
We don't need this method, it can be safely replaced with a oneliner: `connection.getHeaderField("X-Branch-Request-Id");` which does the same thing as the method. However, the string `"X-Branch-Request-Id"` should be pulled out and defined as one of the values in a new enum `Headers` in the `Defines` class.

Reply:
Yeah, missed the constant, will update

Reply:
@code-crusher this seems to be using a String literal still

Reply:
Dunzo